src/PartitionService.ts

```diff
@@ -14,84 +14,73 @@
  * limitations under the License.
  */
 
-import * as Promise from 'bluebird';
 import HazelcastClient from './HazelcastClient';
 import {ILogger} from './logging/ILogger';
-import Address = require('./Address');
-import ClientMessage = require('./ClientMessage');
-import GetPartitionsCodec = require('./codec/GetPartitionsCodec');
+import {ClientConnection} from './network/ClientConnection';
+import {UUID} from './core/UUID';
+import {ClientOfflineError} from './HazelcastError';
 
-const PARTITION_REFRESH_INTERVAL = 10000;
+class PartitionTable {
+    connection: ClientConnection;
+    partitionStateVersion: number = -1;
+    partitions = new Map<number, UUID>();
+}
 
+/**
+ * Partition service for Hazelcast clients.
+ *
+ * Allows to retrieve information about the partition count, the partition owner or the partitionId of a key.
+ */
 export class PartitionService {
 
     private client: HazelcastClient;
-    private partitionMap: { [partitionId: number]: Address } = {};
-    private partitionCount: number;
-    private partitionRefreshTask: any;
-    private isShutdown: boolean;
+    private partitionTable = new PartitionTable();
+    private partitionCount: number = 0;
     private logger: ILogger;
 
     constructor(client: HazelcastClient) {
         this.client = client;
         this.logger = client.getLoggingService().getLogger();
-        this.isShutdown = false;
-    }
-
-    initialize(): Promise<void> {
-        this.partitionRefreshTask = setInterval(this.refresh.bind(this), PARTITION_REFRESH_INTERVAL);
-        return this.refresh();
-    }
-
-    shutdown(): void {
-        clearInterval(this.partitionRefreshTask);
-        this.isShutdown = true;
     }
 
     /**
-     * Refreshes the internal partition table.
+     * The partitions can be empty on the response, client will not apply the empty partition table,
```

Comment:
nit: change the comma to a dot at the line end: `empty partition table,` -> `empty partition table.`